### PR TITLE
Add deactivation note for feature plugin

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #

Adds a note to deactivate older versions of wc admin when the included core version is greater than the feature plugin running.

Notes:
* We can't use a standard URL with a nonce since the notes action is stored in the DB and the nonce would expire.
* We probably need to add some permission checks on notes at some point to prevent notes from being visible to those without permissions.  Currently this will show a permission error after clicking the action if the user does not have permission to disable plugins.

### Screenshots
Screen Shot 2020-02-09 at 1 12 37 PM

### Detailed test instructions:

1. Symlink or copy this branch to the WC 4.0 beta `woocommerce/packages/woocommerce-admin` directory.
1. Run an older version as a feature plugin (or force this by changing the version constant in `Package.php`).
1. Visit any WC page with store alerts.
1. Note the error.
1. Disable the plugin with the action.
1. Make sure the feature plugin is successfully disabled.

### CODE REVIEW

**Comment by rrennick** [comment]
@joshuatf The logic on this works perfectly except that with the autoloaders the `Package.php` autoloaded by WC is the plugin file `plugins/woocommerce-admin/src/Package.php` vs the WC `/packages/...` file. We'll have to look at excluding `Package.php` from the local autoloading.

**Comment by jeffstieler** [comment]
@rrennick is that change required to approve this PR?

**Comment by rrennick** [comment]
> is that change required to approve this PR?

Yes, the notice will never be displayed on any version since `Package.php` was added because it's comparing the version constants in the standalone copy of the plugin.

**Comment by rrennick** [comment]
I have updated the PR by adding `\Compose` to the namespace on the `Package` class. Adding the folder to the namespace prevents the WC Admin autoloader from finding the class file. The WC core autoloader isn't using PSR-4 for autoloading packages, so its autoloader finds and loads the `Package` class in the `packages` folder.

This requires a PR in WC Core which I'll create in a few minutes.

**Comment by rrennick** [comment]
@jeffstieler This one is ready for review.

**Comment by jeffstieler** [comment]
> I have updated the PR by adding `\Compose` to the namespace on the `Package` class. Adding the folder to the namespace prevents the WC Admin autoloader from finding the class file. The WC core autoloader isn't using PSR-4 for autoloading packages, so its autoloader finds and loads the `Package` class in the `packages` folder.

Just to make sure I understand this (and that's usually where I ask for a lengthy comment block explaining the code): the addition of `\Compose` to the namespace (or any additional path) causes the `Package` class to not get autoloaded when any other class in the `\Automattic\WooCommerce\Admin` namespace is referenced?

If I'm understanding that correctly - why "Compose" as the additional path? Was that supposed to be `Composer`?

**Comment by rrennick** [comment]
> to the namespace (or any additional path) causes the Package class to not get autoloaded when any other class in the \Automattic\WooCommerce\Admin namespace is referenced

More or less. The autoloading function loads a class file when that class is referenced. When WC Admin is active as a standalone plugin, the Package class file isn't loaded because the SA plugin doesn't reference it.

> why "Compose" as the additional path? Was that supposed to be Composer?

I was thinking of differentiating between composing the standalone plugin and the composer package. I'd be okay with `Composer` or `Composed`. If the class name were something different then I would have added `Package` to the namespace. `Composed` might signal that it's a "composed package".

File: src/Package.php [hunk 1]
```diff
@@ -39,6 +41,13 @@ public static function init() {
 
 		// Avoid double initialization when the feature plugin is in use.
 		if ( defined( 'WC_ADMIN_VERSION_NUMBER' ) ) {
+			$update_version = new WC_Admin_Notes_Update_Version();
+			if ( version_compare( WC_ADMIN_VERSION_NUMBER, self::VERSION, '<' ) ) {
+				$update_version::add_note();
+			} else {
+				$update_version::delete_note();
```
Comment (psealock) [inline src/Package.php]: Why not just have the `if` statement and not create the note at all in the `else` case.

Comment (rrennick) [reply src/Package.php]: The user could update WCA to the current version. Without the delete_note, the notice would continue to display.

File: src/Notes/WC_Admin_Notes_Update_Version.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+namespace Automattic\WooCommerce\Admin\Notes;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WC_Admin_Notes_Update_Version.
+ */
+class WC_Admin_Notes_Update_Version {
+	const NOTE_NAME = 'wc-admin-update-version';
```
Comment (psealock) [inline src/Notes/WC_Admin_Notes_Update_Version.php]: Should this be `wc-deactivate-outdated-plugin` to be more reflective of the call to action?

Comment (rrennick) [reply src/Notes/WC_Admin_Notes_Update_Version.php]: Went with `wc-admin-deactivate-plugin`.

**Comment by psealock** [comment]
https://github.com/woocommerce/woocommerce-admin/pull/3687/commits/f29854925217946e4c37fa5713cf78be0d53fc35 makes the renamed note name the class and filename for consistency.

File: src/Package.php [hunk 3]
```diff
@@ -39,6 +46,13 @@ public static function init() {
 
 		// Avoid double initialization when the feature plugin is in use.
 		if ( defined( 'WC_ADMIN_VERSION_NUMBER' ) ) {
+			$update_version = new WC_Admin_Notes_Deactivate_Plugin();
+			if ( version_compare( WC_ADMIN_VERSION_NUMBER, self::VERSION, '<' ) ) {
+				$update_version::add_note();
+			} else {
+				$update_version::delete_note();
```
Comment (peterfabian) [inline src/Package.php]: If the `WC_ADMIN_VERSION_NUMBER` is not defined, that means the plugin is either inactive or not installed, so the notice should not be displayed. 
However, in my testing with WC Admin 0.23.3, the code never hits `$update_version::delete_note();`.

**Pull request merged**